src/Chapter4.hs

```diff
@@ -488,7 +503,22 @@ Implement the 'Applicative' instance for our 'List' type.
   may also need to implement a few useful helper functions for our List
   type.
 -}
+{- ORMOLU_ENABLE -}
+instance Applicative List where
+  pure :: a -> List a
+  pure x = Cons x Empty
+
+  (<*>) Empty _ = Empty
+  (<*>) fs ms =
+    goM Empty ms
+    where
+      goF acc Empty _ = acc
+      goF acc (Cons f fTail) m = goF (Cons (f m) acc) fTail m
 
+      goM acc Empty = acc
+      goM acc (Cons m mTail) = goF (goM acc mTail) fs m
```

Comment:
I would strongly advise adding type signatures to local functions (`goF` and `goM` in this case). For example, after adding a type signature you may notice some similarities with existing functions and probably assign clearer names 🙂 Current implementation is a bit difficult to read because of existing names.

If you have only a single recursive function in `where`, it's okay to call it `go`, as you already have the name for the top-level function, so no need to create an extra name for recursive helper.

Reply:
I suppose one should do `flatten` another `map` lol, but can't find a way to make it